Can't add a solution target BeforeTargets ValidateSolutionConfiguration

[Based on an offline conversation with @bruno-garcia]

It should be possible to add a target that runs before MSBuild's solution handling code tries to build any project. `ValidateSolutionConfiguration` is a good hook for that: the first target run in a solution-driven build. But it doesn't work.

Minimal repro: [msbuild6452.zip](https://github.com/dotnet/msbuild/files/6474378/msbuild6452.zip). It should always emit the message, but instead it complains that the project doesn't exist.

```sh-session
❯ msbuild .\Sentry.Unity.sln
Microsoft (R) Build Engine version 16.10.0-preview-21256-02+c040391f9 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

Building the projects in this solution one at a time. To enable parallel build, please add the "-m" switch.
Build started 5/13/2021 1:15:39 PM.
Project "S:\work\brunogarcia\Sentry.Unity.sln" on node 1 (default targets).
ValidateSolutionConfiguration:
  Building solution configuration "Debug|Any CPU".
S:\work\brunogarcia\Sentry.Unity.sln.metaproj : error MSB3202: The project file "S:\work\brunogarcia\ProjectThatDoesNotExist\ProjectThatDoesNotExist.csproj" was not found. [S:\work\brunogarcia\Sentry.Unity.sln]
Done Building Project "S:\work\brunogarcia\Sentry.Unity.sln" (default targets) -- FAILED.


Build FAILED.

"S:\work\brunogarcia\Sentry.Unity.sln" (default target) (1) ->
(Build target) ->
  S:\work\brunogarcia\Sentry.Unity.sln.metaproj : error MSB3202: The project file "S:\work\brunogarcia\ProjectThatDoesNotExist\ProjectThatDoesNotExist.csproj" was not found. [S:\work\brunogarcia\Sentry.Unity.sln]

    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:00.13
```


## Workaround

Set your custom target as an InitialTargets like so

```xml
<Project InitialTargets="RestoreSubmodule">
  <Target Name="RestoreSubmodule">
    <Message Importance="High" Text="Did the thing super early in the solution build"></Message>
  </Target>
</Project>
```

--- a/src/Build/Construction/Solution/SolutionProjectGenerator.cs
+++ b/src/Build/Construction/Solution/SolutionProjectGenerator.cs
@@ -66,7 +66,11 @@ namespace Microsoft.Build.Construction
             "Build",
             "Clean",
             "Rebuild",
-            "Publish"
+            "Publish",
+            "ValidateSolutionConfiguration",
+            "ValidateToolsVersions",
+            "ValidateProjects",
+            "GetSolutionConfigurationContents"
             );
 
 #if FEATURE_ASPNET_COMPILER

